build-tools/packages/build-cli/src/machines/fluidReleaseMachine.ts

```diff
@@ -0,0 +1,710 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { strict as assert } from "assert";
+import path from "path";
+import { Context, exec, FluidRepo, MonoRepo, MonoRepoKind } from "@fluidframework/build-tools";
+import { bumpVersionScheme, detectVersionScheme } from "@fluid-tools/version-tools";
+import { FileSystem as fs } from "@rushstack/node-core-library";
+import chalk from "chalk";
+import inquirer from "inquirer";
+import { from as createStateMachine, Machine } from "jssm";
+import { isReleaseGroup } from "../releaseGroups";
+import { CommandLogger } from "../logging";
+import { InstructionalPrompt } from "../instructionalPromptWriter";
+import {
+    getPreReleaseDependencies,
+    isReleased,
+    bumpReleaseGroup,
+    npmCheckUpdates,
+    difference,
+    generateReleaseBranchName,
+    generateBumpVersionBranchName,
+    generateBumpDepsBranchName,
+} from "../lib";
+import { MachineState } from "./machineState";
+import { BaseStateHandler, InitFailedStateHandler } from "./stateHandlers";
+import { FluidReleaseStateHandlerData } from "./fluidReleaseStateHandlerData";
+import { askForReleaseType, checkPolicy, checkShouldRunOptionalChecks } from "./handlerFunctions";
+
+// eslint-disable-next-line unicorn/prefer-module
+const machineDefinitionFile = path.join(__dirname, "FluidRelease.fsl");
+const fsl = fs.readFile(machineDefinitionFile).toString();
+
+/**
+ * An FSL state machine that encodes the Fluid release process.
+ *
+ * @alpha
+ */
+export const FluidReleaseMachineDefinition = createStateMachine(fsl);
+
+export class FluidReleaseStateHandler extends InitFailedStateHandler {
+    private static async succeedAndContinue(
+        context: Context,
+        state: MachineState,
+        machine: Machine<unknown>,
+        testMode: boolean,
+        log: CommandLogger,
+        data: FluidReleaseStateHandlerData,
+    ): Promise<boolean> {
+        if (testMode) return true;
+
+        BaseStateHandler.signalSuccess(machine, state);
+        return true;
+    }
+
+    // eslint-disable-next-line complexity, max-params
+    async handleState(
+        state: MachineState,
+        machine: Machine<unknown>,
+        testMode: boolean,
+        log: CommandLogger,
+        data: FluidReleaseStateHandlerData,
+    ): Promise<boolean> {
+        const { context } = data;
+        assert(context !== undefined, "Context is undefined.");
+
+        let superShouldHandle = false;
+
+        switch (state) {
+            case "AskForReleaseType": {
+                await askForReleaseType(state, machine, testMode, log, data);
+                break;
+            }
+
+            case "CheckShouldRunOptionalChecks": {
+                await checkShouldRunOptionalChecks(state, machine, testMode, log, data);
+                break;
+            }
```

Comment:
@Josmithr I'm starting to move the individual state handlers to their own pure functions all with the same signature. Before I continue I wanted to get your input on this pattern and make sure it aligned with your feedback from my previous PR.

Reply:
Yeah, I think this approach is a lot easier to parse!